From 7813adf66be31d8b0e8df21821e786e688f7fe78 Mon Sep 17 00:00:00 2001
From: Bryce Harrington <bryce@bryceharrington.org>
Date: Fri, 27 Mar 2009 19:01:32 -0700
Subject: [PATCH] cwGetBackingPicture can segfault when minimizing/maximizing firefox with
 a flash video playing.  This appears to be a race condition in which the
 backing picture's data is not always fully defined.

Signed-off-by: Bryce Harrington <bryce@bryceharrington.org>
---
 miext/cw/cw_render.c |   13 +++++++++----
 1 files changed, 9 insertions(+), 4 deletions(-)

Index: xorg-server/miext/cw/cw_render.c
===================================================================
--- xorg-server.orig/miext/cw/cw_render.c	2011-04-09 10:36:56.000000000 +0300
+++ xorg-server/miext/cw/cw_render.c	2011-04-09 10:38:10.000000000 +0300
@@ -123,10 +123,15 @@
 	WindowPtr   pWindow = (WindowPtr) pDrawable;
 	PixmapPtr   pPixmap = getCwPixmap (pWindow);
 
-	*x_off = pDrawable->x - pPixmap->screen_x;
-	*y_off = pDrawable->y - pPixmap->screen_y;
+        if (pDrawable && pPixmap) {
+            *x_off = pDrawable->x - pPixmap->screen_x;
+            *y_off = pDrawable->y - pPixmap->screen_y;
 
-	return pPicturePrivate->pBackingPicture;
+            return pPicturePrivate->pBackingPicture;
+        } else {
+            *x_off = *y_off = 0;
+            return pPicture;
+        }
     }
     else
     {
